### PR TITLE
Fix storelogs

### PR DESCRIPTION
**Pull request by toshiiw**
This PR ensures worker logs are stored correctly.
Also tweak DB table definitions for consistency.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/222)

### CODE REVIEW

**Comment by YujiOshima** [comment]
@toshiiw Thanks!
related to #220 
/ok-to-test

File: pkg/db/interface.go [hunk 1]
```diff
@@ -455,18 +455,60 @@ func (d *dbConn) GetWorkerLogs(id string, opts *GetWorkerLogOpts) ([]*WorkerLog,
...
-		row := d.db.QueryRow("SELECT time, value FROM worker_lastlogs WHERE worker_id = ?", id)
-		err = row.Scan(&lastTimestamp, value)
-	} else {
-		row := d.db.QueryRow("SELECT time FROM worker_lastlogs WHERE worker_id = ?", id)
-		err = row.Scan(&lastTimestamp)
+	// Use LEFT JOIN to ensure a result even if there's no matching
+	// in worker_metrics.
+	rows, err := d.db.Query(
+		"SELECT worker_lastlogs.time, name, value FROM worker_lastlogs "+
+			"LEFT JOIN worker_metrics ON (worker_lastlogs.worker_id = worker_metrics.worker_id AND worker_lastlogs.time = worker_metrics.time) "+
```
Comment (YujiOshima) [inline pkg/db/interface.go]: Why not using ` to make a multiple line string?

Comment (toshiiw) [reply pkg/db/interface.go]: Fixed.

File: pkg/db/db_init.go [hunk 2]
```diff
@@ -54,21 +54,22 @@ func (d *dbConn) DBInit() {
...
-		"value TEXT, " +
-		"is_objective TINYINT)")
+	_, err = db.Exec(`CREATE TABLE IF NOT EXISTS worker_metrics
+		(worker_id CHAR(16) NOT NULL,
+		id INT AUTO_INCREMENT PRIMARY KEY,
+		time DATETIME(6),
+		name VARCHAR(255),
+		value TEXT,
+		is_objective TINYINT,
+		FOREIGN KEY (worker_id) REFERENCES workers(id) ON DELETE CASCADE)`)
```
Comment (YujiOshima) [inline pkg/db/db_init.go]: Could you replace `" +` to ` on other tables?

Comment (toshiiw) [reply pkg/db/db_init.go]: My idea was to minimize the size of diff. It might be better to tidy all the things now. Just a sec.

**Comment by YujiOshima** [comment]
@toshiiw Great, Thanks!
/lgtm
/approve

**Pull request merged**